corehq/motech/openmrs/repeater_helpers.py

```diff
@@ -415,14 +415,46 @@ def get_patient_by_uuid(requests, uuid):
     return requests.get('/ws/rest/v1/patient/' + uuid, {'v': 'full'}).json()
 
 
-def get_patient_by_id(requests, patient_identifier_type, patient_identifier):
-    if patient_identifier_type == PERSON_UUID_IDENTIFIER_TYPE_ID:
-        patient = get_patient_by_uuid(requests, patient_identifier)
-        return patient
+def get_patient_by_identifier(requests, identifier_type_uuid, value):
+    """
+    Return the patient that matches the given identifier. If the
+    number of matches is zero or more than one, return None.
+    """
+    response_json = search_patients(requests, value)
+    patients = []
+    for patient in response_json['results']:
+       for identifier in patient['identifiers']:
+            if (
+                identifier['identifierType']['uuid'] == identifier_type_uuid and
+                identifier['identifier'] == value
+            ):
+                patients.append(patient)
+    try:
+        patient, = patients
+    except ValueError:
+        return None
     else:
-        response_json = search_patients(requests, patient_identifier)
-        return PatientSearchParser(response_json).get_patient_matching_identifiers(
-            patient_identifier_type, patient_identifier)
+        return patient
+
+
+def get_patient_by_id(requests, patient_identifier_type, patient_identifier):
```

Comment:
(Changes copied from #20035 to avoid a merge conflict.) Only line 448 is different: 
```python
    return get_patient_by_identifier(requests, patient_identifier_type, patient_identifier)
```